### PR TITLE
feat: show audio attachments in message list

### PR DESCRIPTION
**Pull request by khushal87**
## 🎯 Goal

The goal of the PR is to render the audio attachments within the message attachments within the message list.

## 🛠 Implementation details

There's nothing much here except dealing with the case of rendering the logic for audio attachment in FileAttachment.

## 🎨 UI Changes
simulator_screenshot_4B9B97B4-4DD3-4A9E-8416-FB1255AB250B

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Documentation is updated
- [x] New code is tested in main example apps, including all possible scenarios
  - [x] SampleApp iOS and Android
  - [x] Expo iOS and Android

### CODE REVIEW

File: examples/ExpoMessaging/ios/ExpoMessaging.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -388,7 +388,7 @@
 				COPY_PHASE_STRIP = NO;
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
 				ENABLE_TESTABILITY = YES;
-				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "";
+				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "arm64 ";
```
Comment (santhoshvai) [inline examples/ExpoMessaging/ios/ExpoMessaging.xcodeproj/project.pbxproj:391]: why was this needed? any clue?

File: package/src/components/Attachment/AudioAttachment.tsx [hunk 2]
```diff
@@ -328,35 +329,29 @@ const AudioAttachmentUploadPreviewWithContext = <
...
-export type AudioAttachmentUploadPreviewProps<
+export type AudioAttachmentProps<
   StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
-> = Partial<AudioAttachmentUploadPreviewPropsWithContext<StreamChatGenerics>> & {
+> = Partial<AudioAttachmentPropsWithContext<StreamChatGenerics>> & {
   index: number;
-  item: FileUpload;
+  item: Omit<FileUpload, 'state'>;
   onLoad: (index: string, duration: number) => void;
   onPlayPause: (index: string, status?: boolean) => void;
```
Comment (santhoshvai) [inline package/src/components/Attachment/AudioAttachment.tsx:338]:
```suggestion
  onPlayPause: (index: string, currentlyPlayed?: boolean) => void;
```

the status seems too generic here and non understandable.

File: examples/ExpoMessaging/ios/ExpoMessaging.xcodeproj/project.pbxproj (hunk 1 above)
Comment (khushal87) [reply examples/ExpoMessaging/ios/ExpoMessaging.xcodeproj/project.pbxproj:391]: Sorry, I committed it unwaveringly and will fix it.

File: package/src/components/Attachment/AudioAttachment.tsx (hunk 2 above)
Comment (khushal87) [reply package/src/components/Attachment/AudioAttachment.tsx]: Fixed, looking for your final review and merge! Thanks 😄

**Review by santhoshvai — APPROVED** [review]

**Review by santhoshvai — APPROVED** [review]
LGTM.. but consider my suggestions if applicable

File: package/src/components/Attachment/FileAttachmentGroup.tsx [hunk 3]
```diff
@@ -35,12 +43,73 @@ export type FileAttachmentGroupPropsWithContext<
...
+              : currentTime
+              ? currentTime / (fileUpload.duration as number)
+              : 0
+            : fileUpload.progress,
+      })),
+    );
+  };
+
+  // The handler which controls or sets the paused/played state of the audio.
+  const onPlayPause = (index: string, status?: boolean) => {
```
[the remark this answers is not in the document]
Comment (khushal87) [reply package/src/components/Attachment/FileAttachmentGroup.tsx]: This is fixed by changing to `pausedStatus`.
